From 8694fd1cdbc687a3d70a595a3cb0077d335df4bb Mon Sep 17 00:00:00 2001
From: Ranjani Vaidyanathan <ra5478@freescale.com>
Date: Tue, 11 Dec 2012 17:50:33 -0600
Subject: ENGR00236837 MX6SL-Fix random crash caused by incorrect setting of
 IPG clk rate.

Need to ensure that bus frequency setpoint is changed only if
the system is not already at the requested setpoint.
Changing the bus freq to high setpoint when its already at
high setpoint causes the AHB/IPG dividers to be set incorrectly.
Then when the system enters WAIT mode, the 12:5 ratio of
ARM_CLK:IPG_CLK is no longer maintained.
This causes random crashes.
Fix is to return immediately if the bus is already at the
requested setpoint

Signed-off-by: Ranjani Vaidyanathan <ra5478@freescale.com>
---
 arch/arm/mach-mx6/bus_freq.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mach-mx6/bus_freq.c b/arch/arm/mach-mx6/bus_freq.c
index 6422052..1f429f9 100644
--- a/arch/arm/mach-mx6/bus_freq.c
+++ b/arch/arm/mach-mx6/bus_freq.c
@@ -300,6 +300,8 @@ int set_high_bus_freq(int high_bus_freq)
 	if (!bus_freq_scaling_initialized || !bus_freq_scaling_is_active)
 		return 0;
 
+	if (cpu_is_mx6sl())
+		high_bus_freq = 1;
 
 	if (high_bus_freq_mode && high_bus_freq)
 		return 0;
-- 
1.7.9.5

